src/storm/generator/PrismNextStateGenerator.cpp

```diff
@@ -592,6 +593,22 @@ PrismNextStateGenerator<ValueType, StateType>::getActiveCommandsByActionIndex(ui
     return result;
 }
 
+template<typename ValueType, typename BaseValueType>
+ValueType evaluateLikelihoodExpression(storm::prism::Update const& update, storm::expressions::ExpressionEvaluator<BaseValueType> const& evaluator) {
+    if constexpr (storm::IsIntervalType<ValueType>) {
+        if (update.isLikelihoodInterval()) {
+            BaseValueType lower = evaluator.asRational(update.getLikelihoodExpressionInterval().first);
+            BaseValueType upper = evaluator.asRational(update.getLikelihoodExpressionInterval().second);
+            return ValueType(lower, carl::BoundType::WEAK, upper, carl::BoundType::WEAK);
+        } else {
+            return evaluator.asRational(update.getLikelihoodExpression());
+        }
+    } else {
+        STORM_LOG_ASSERT(!update.isLikelihoodInterval(), "Unexpected interval likelihood for non-interval model in update " << update << ".");
```

Comment:
We should probably also warn about this with exploration checks on.

Reply:
If this assertion triggers, it would really be a bug in the code logic as we already check whether there are intervals in the PRISM program during the construction (line 44-45):

```cpp
    STORM_LOG_THROW(storm::IsIntervalType<ValueType> || !this->program.hasIntervalUpdates(), storm::exceptions::WrongFormatException,
                    "The provided PRISM program contains interval updates, which are not supported by the chosen value type.");

```

In my view, this type of bug is not what exploration checks are supposed to handle.